src/json.c

```diff
@@ -459,38 +621,35 @@ int json_nvmet_view_ports(struct NVMET_PORTS *ports)
  * }
  */
 
-int json_status_check(unsigned char *message)
+/**
+ * It creates a JSON object with two key/value pairs, "status" and "version", and returns the JSON object as a string
+ *
+ * @param json_str This is a pointer to a pointer to a char.  It's used to return the JSON string to the caller.
+ *
+ * @return An @p int representing the result.
+ */
+int json_status_check(char **json_str)
 {
+	int rc = INVALID_VALUE;
 	json_t *root = json_object();
 
 	json_object_set_new(root, "status", json_string("OK"));
 	json_object_set_new(root, "version", json_string(VERSION_NUM));
 
-	sprintf(message, "%s\n", json_dumps(root, 0));
-
-	json_decref(root);
-
-	return SUCCESS;
-}
-
-/*
- * JSON output format:
- * {
- *    "status": "Unsupported"
- * }
- */
-
-int json_status_unsupported(unsigned char *message)
-{
-	json_t *root = json_object();
-
-	json_object_set_new(root, "status", json_string("Unsupported"));
-
-	sprintf(message, "%s\n", json_dumps(root, 0));
+	char *jdumped = json_dumps(root, 0);
+	if (jdumped != NULL) {
+		if (json_str != NULL) {
+			*json_str = jdumped;
+			rc = SUCCESS;
+		}
+//		sprintf(message, "%s\n", jdumped);
```

Comment:
Again,  it looks aa if this was used for troubleshooting and if we do not need it, we should remove the commented code.

Reply:
Removed, fixed.